internal/pkg/build/sources/packer_sif_linux.go

```diff
@@ -33,73 +34,69 @@ func (p *SIFPacker) Pack() (*types.Bundle, error) {
 
 // First pass just assumes a single system partition, later passes will handle more complex sif files
 // unpackSIF parses through the sif file and places each component in the sandbox
-func (p *SIFPacker) unpackSIF(b *types.Bundle, rootfs string) (err error) {
-
-	// load the container
-	fimg, err := sif.LoadContainer(rootfs, true)
-	if err != nil {
-		sylog.Errorf("error loading sif file %s: %s\n", rootfs, err)
-		return err
-	}
-	defer fimg.UnloadContainer()
-
-	// Get the default system partition image as rootfs part
-	part, _, err := fimg.GetPartPrimSys()
+func (p *SIFPacker) unpackSIF(b *types.Bundle, srcfile string) (err error) {
+	img, err := image.Init(srcfile, false)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not open image %s: %s", srcfile, err)
 	}
-
-	// record the fs type
-	mountType := ""
-	fstype, err := part.GetFsType()
-	if err != nil {
-		return err
-	}
-	if fstype == sif.FsSquash {
-		mountType = "squashfs"
-	} else if fstype == sif.FsExt3 {
-		mountType = "ext3"
-	} else {
-		return fmt.Errorf("unknown file system type: %v", fstype)
-	}
-
-	info := &loop.Info64{
-		Offset:    uint64(part.Fileoff),
-		SizeLimit: uint64(part.Filelen),
-		Flags:     loop.FlagsAutoClear,
-	}
-
-	//copy partition contents to bundle rootfs
-	err = unpackImagePartion(fimg.Fp.Name(), b.Rootfs(), mountType, info)
-	if err != nil {
-		return fmt.Errorf("While copying partition data to bundle: %v", err)
+	defer img.File.Close()
+
```

Comment:
safest to wait #3195 merge to add a `if !img.HasRootFs()`

Reply:
Will do